tests/covariance_matrix_04.cc

```diff
@@ -0,0 +1,70 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the SampleFlow authors.
+//
+// This file is part of the SampleFlow library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Check the CovarianceMatrix consumer with complex numbers
+
+
+#include <iostream>
+#include <fstream>
+#include <valarray>
+#include <complex>
+
+#include <sampleflow/producers/metropolis_hastings.h>
+#include <sampleflow/consumers/covariance_matrix.h>
+
+using ComponentType = std::complex<double>;
+using SampleType = std::valarray<ComponentType>;
+
+double log_likelihood (const SampleType &x)
+{
+  return 1;
+}
+
+std::pair<SampleType,double> perturb (const SampleType &x)
+{
+  SampleType y = x;
+
+  for (auto &el : y)
+    y += ComponentType(1, 1);
+
+  // Return both the new sample and the ratio of proposal distribution
+  // probabilities. We've moved the sample to the right, so that ratio
+  // is actually infinity, but we can lie about it for the purposes of
+  // this test.
+  return {y, 1.0};
+}
+
+int main ()
+{
+
+  SampleFlow::Producers::MetropolisHastings<SampleType> mh_sampler;
+
+  SampleFlow::Consumers::CovarianceMatrix<SampleType> covariance_matrix;
+  covariance_matrix.connect_to_producer(mh_sampler);
+
+  mh_sampler.sample ({ComponentType(0, 0), ComponentType(1, 1)},
+                     &log_likelihood,
+                     &perturb,
+                     8);
+
+  // At this point, we have sampled two sets: first starts from 1 and
+  // goes to 8, second - starts from 2 and goes to 9. Sample variance
+  // for both vector is 6 as covariance too. Output whatever we got:
+  std::cout << covariance_matrix.get()(0,0) << std::endl;
```

Comment:
This used to be true for the _02 test, but apparently you now get 48+0i. Have you verified that 48 is the correct answer? If so, update the comment here.